celestia/static/run_node.js

```diff
@@ -0,0 +1,74 @@
+Error.stackTraceLimit = 99; // rust stack traces can get pretty big, increase the default
+
+import init, { setup_logging, Network, WasmNode, WasmNodeConfig, canonical_network_bootnodes, network_genesis } from "/wasm/wasm_node.js";
+
+async function fetch_config() {
+    const response = await fetch('/cfg.json');
+    const json = await response.json();
+
+    console.log("Received config:", json);
+
+    const network = json.network;
+    const bootnodes = json.bootnodes
+    if (bootnodes.length === 0) {
+        bootnodes.push(...canonical_network_bootnodes(network));
+    }
+    const genesis = network_genesis(network);
+
+    return new WasmNodeConfig(network, genesis, bootnodes);
+}
+
+async function show_stats(node) {
+    if (!node) {
+        return;
+    }
+    document.getElementById("syncer").innerText = JSON.stringify(await node.syncer_info());
+
+    let peers_ul = document.createElement('ul');
+    (await node.connected_peers()).forEach(function(peer) {
+        var li = document.createElement("li");
+        li.innerText = peer;
+        peers_ul.appendChild(li);
+    });
+
+    document.getElementById("peers").replaceChildren(peers_ul);
+}
+
+
+function bind_config() {
+    // TODO two way binding between window.config and input values
+}
+
++async function main(document, window, undefined) {
```

Comment:
What is the `+` here?

Reply:
I think the `run_node.js` is a bit drafty now :)

Reply:
🤯